test/functional/p2p_add_connections.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test add_outbound_p2p_connection test framework functionality"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.p2p import P2PInterface
+from test_framework.util import assert_equal
+
+
+class P2PAddConnections(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def check_node_connections(self, node=None, num_in=None, num_out=None):
```

Comment:
You always call this function with all arguments, so no need to have defaults (and in fact the function would fail if you called it without arguments because `self.nodes[None]` is a TypeError.

Reply:
You could also change this to be a standalone function rather than a method in the P2PAddConnections class and just take a node as an argument rather than an index:

```python
def check_node_connections(node, num_in, num_out):
    info = node.getnetworkinfo()
    assert_equal(info["connections_in"], num_in)
    assert_equal(info["connections_out"], num_out)
```

Reply:
Agree with @jnewbery's suggestions. @MarcoFalke also reminded me recently in https://github.com/bitcoin/bitcoin/pull/20403#discussion_r530307834 that you can enforce use of named args with the splat. [Example](https://github.com/bitcoin/bitcoin/pull/20391/files#diff-5450eea6e8ed6364028357e2f1abc3dfcf458177655f518336266f08190f0dcfR323).
```suggestion
    def check_node_connections(*, node, num_in, num_out):
```

Reply:
ah, yes I put in the defaults to try to enforce the named args. cool splat trick @jonatack.

also converted to standalone function